src/managers/prompt.ts

```diff
@@ -0,0 +1,886 @@
+      }
+
+      .command .command-highlight {
+        font-weight: bold;
+      }
+
+      .command-Tip {
+        color: #5a5a5a;
+        text-align: center;
+        padding: 12px 12px;
+        font-size: 18px;
+      }
+      
+      .current-value {
+        background-color: #a7b8c1;
+        color: white;
+        border-radius: 5px;
+        padding: 0 5px;
+        margin-left: 10px;
+        font-size: 14px;
+        vertical-align: middle;
+        letter-spacing: 0.05em;
+      }
+
+      /* instructions */
+      .prompt-instructions {
+        display: flex;
+        align-content: center;
+        justify-content: center;
+        font-size: 15px;
+        color: rgba(0, 0, 0, 0.4);
+        height: 2.5em;
+        width: 100%;
+        border-top: 1px solid #f6f6f6;
+        margin-top: 5px;
+      }
+      
+      .prompt-instruction {
+        margin: auto .5em;  
+      }
+      
+      .prompt-instruction-command {
+        margin-right: .1em;
+        font-weight: 600;
+      }
+    `
+    this.document.querySelector('#main-window').appendChild(style)
+
+  }
+
+  private createElement(nodeName: string): HTMLElement {
+    return this.document.createElementNS("http://www.w3.org/1999/xhtml", nodeName)
+  }
+
+  private removeKeys() {
+    if (this.keyset) {
+      this.keyset.remove()
+    }
+  }
+
+  private initKeys() {
+    this.removeKeys()
+    let keyset = this.document.createElement("keyset") as XUL.Element;
+    keyset.setAttribute("id", "toolkit-keyset");
+
+    let key = this.document.createElement("key");
+    key.setAttribute("id", "toolkit-prompt-key");
+    key.setAttribute("oncommand", "console.log(1)");
+    key.addEventListener("command", function () {
+      let Zotero =
+        Components.classes["@zotero.org/Zotero;1"].getService(
+          Components.interfaces.nsISupports
+        ).wrappedJSObject;
+      let promptNode = document.querySelector(".prompt") as HTMLDivElement
+      if (promptNode.style.display == "none") {
+        promptNode.style.display = "flex"
+        promptNode.querySelector("input").focus()
+        if (Zotero._Prompt.path.length == 0) {
+          Zotero._Prompt.refresh()
+        }
+      } else {
+        promptNode.style.display = "none"
+      }
+    })
+    key.setAttribute("key", "p")
+    key.setAttribute("modifiers", "shift")
+    keyset.appendChild(key)
+    this.keyset = keyset
+    this.document.getElementById("mainKeyset").parentNode.appendChild(keyset);
+  }
+}
+
+export class PromptManager extends ManagerTool {
+  /**
+   * Prompt object
+   */
+  public prompt: Prompt;
+  /**
+   * Save the commands registered by this manager
+   */
+  public commands: Command[] = []
+  constructor(base?: BasicTool | BasicOptions) {
+    super(base);
+    if (!Zotero._Prompt) {
+      Zotero._Prompt = new Prompt()
+    }
+    this.prompt = Zotero._Prompt
+  }
+
+  /**
+   * An example of registering commads.
+   */
+  public registerExample() {
```

Comment:
Put examples in comment, start with @example, or add it to plugin template later.